src/Symfony/Component/RateLimiter/AbstractRequestRateLimiter.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\RateLimiter;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * A special type of compound limiter that deals with requests.
+ *
+ * This allows to limit on different types of information
+ * from the requests.
+ *
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ *
+ * @experimental in Symfony 5.2
+ */
+abstract class AbstractRequestRateLimiter
```

Comment:
Should it implement `ResetInterface?` (+ `kernel.reset` tag on the definition?)

Reply:
I've looked at the `ResetInterface` (also when creating the `LimiterInterface` - which this class' methods are based on), but the PHPdocs say "This method could be called, for example, to recycle objects that are used as services, so that they can be used to handle several requests in the same process loop (note that we advise making your services stateless instead of implementing this interface when possible.)".

(Request) Rate limiters should not be reset for each request or in a process loop. The only real use-case I've found yet is that of login throttling: resetting the counter upon succesful login.

Reply:
Hmm, this got me thinking: I think we should drop the reset functionality. This would mean you can still easily enumerate username + password combinations by creating an account for that service. As long as you add a valid login every 5 attempts, you have infinite login attempts per minute.

Reply:
Fair enough. Agreed for dropping `reset`